src/app_service/service/message/async_tasks.nim

```diff
@@ -295,3 +293,42 @@ const asyncGetMessageByMessageIdTask: Task = proc(argEncoded: string) {.gcsafe,
     }
     arg.finish(output)
 
+#################################################
+# Async mark message as unread
+#################################################
+
+type
+    AsyncMarkMessageAsUnreadTaskArg = ref object of QObjectTaskArg
+      requestId*: string
+      messageId*: string
+      chatId*: string
+
+const asyncMarkMessageAsUnreadTask: Task = proc(argEncoded: string) {.gcsafe, nimcall.} =
+  let arg = decode[AsyncMarkMessageAsUnreadTaskArg](argEncoded)
+
+  var responseJson = %*{
+    "chatId": arg.chatId,
+    "messageId": arg.messageId,
+    "requestId": arg.requestId,
+    "messagesCount": 0,
+    "messagesWithMentionsCount": 0,
+    "error": ""
+  }
+
+  try:
+    let response = status_go.markMessageAsUnread(arg.chatId, arg.messageId)
+
```

Comment:
I think we should also process `response.error` here. That would be the `error` returned by status-go.

Reply:
Added code to process the error